Add instruction on how to use local docker registry (#2076)
knative/docs: docs/serving/getting-started-knative-app.md

## Sample application

This guide demonstrates the basic workflow for deploying the
[Hello World sample app (Go)](../serving/samples/hello-world/helloworld-go) from the 
[Google Container Registry](https://cloud.google.com/container-registry/docs/pushing-and-pulling).
You can use these steps as a guide for deploying your own container images from other
registries like [Docker Hub](https://docs.docker.com/docker-hub/repos/).

In case your image is in the local registry you will want to have Knative skip resolving the image tag. To do this you will need to tag it as following first:

```bash
docker tag local-image dev.local/local-image # use dev.local/local-image in your yamls
```

> 

Reviewer: Add link to more info about image tags
```suggestion

[Learn more about image tag resolution.](./tag-resolution.md)
```

Author: I committed the link as well but this might not be necessary for the reason you already mentioned above.

[Google Container Registry](https://cloud.google.com/container-registry/docs/pushing-and-pulling),
or another container image registry.

> In case your image is in the local registry, you need to tag it as following first:

Reviewer 2: ```suggestion
In case your image is in the local registry you will want to have Knative skip resolving the image tag. To do this you will need to tag it as following first:
```

Author: I commited your suggestion but now it's in own separate commit. Is this ok? Can you squash this or do I have to modify this?

> In case your image is in the local registry you will want to have Knative skip resolving the image tag. To do this you will need to tag it as following first:

Reviewer: ```suggestion
To deploy a local container image, you need to disable image tag resolution by running the following command:
```

```bash
docker tag local-image dev.local/local-image # use dev.local/local-image in your yamls
```

> 

Reviewer 2: It might be helpful to point to the configmap to customize the registry prefix? Or do you think this will be too much information to get started?

Author: Yeah, I think this would be too much for this page. Perhaps it'd be a good idea to have a single page documents these small configurations.

Reviewer: FYI: I think this new text is also too much info for this context but given that we lack a dedicated page or user task that explains how to deploy an image (ie. from a container image registry), its okay for it to live here for now. 

Ideally, we cover only the common used case in a "getting started" to demonstrate a single path to success (and avoid distracting and/or confusing new uses with a bunch of conditionals). If we did have a dedicated topic about deploying from registries, we would instead just have a pointer to that info. For example: 

`[Learn more about deploying your app, including how to deploy local images.](link-to-deploying-topic.md)`

I will keep an eye on this content and implement the suggested changes when we do get around to authoring a stand-alone "how to deploy apps" topic.

> docker tag local-image dev.local/local-image # use dev.local/local-image in your yamls

Reviewer: ```suggestion
# Set to dev.local/local-image when deploying local container images
docker tag local-image dev.local/local-image
```
